tests/components/zha/test_config_flow.py

```diff
@@ -1,125 +1,237 @@
+    )
+
+    with patch.dict(
+        config_flow.RADIO_TYPES,
+        {radio_type: {CONTROLLER: app_ctrl_cls, "radio_description": "radio"}},
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={zigpy.config.CONF_DEVICE_PATH: "/dev/ttyUSB33"},
         )
-        assert mock_radio.connect.call_count == 0
-        assert bad_radio.connect.call_count == 1
-        assert mock_ctrl.startup.call_count == 0
-        assert mock_ctrl.shutdown.call_count == 0
-
-        # successful radio connect
-        assert await config_flow.check_zigpy_connection(
-            mock.sentinel.usb_path, mock.sentinel.radio, mock.sentinel.zigbee_db
+
+        assert result["type"] == "create_entry"
+        assert result["title"].startswith("/dev/ttyUSB33")
+        assert (
+            result["data"][zigpy.config.CONF_DEVICE][zigpy.config.CONF_DEVICE_PATH]
+            == "/dev/ttyUSB33"
         )
-        assert mock_radio.connect.call_count == 1
-        assert bad_radio.connect.call_count == 1
-        assert mock_ctrl.startup.call_count == 1
-        assert mock_ctrl.shutdown.call_count == 1
+        assert result["data"][CONF_RADIO_TYPE] == radio_type
+
+
+def test_get_serial_by_id_no_dir():
+    """Test serial by id conversion if there's no /dev/serial/by-id."""
+    p1 = patch("os.path.isdir", MagicMock(return_value=False))
+    p2 = patch("os.scandir")
+    with p1 as is_dir_mock, p2 as scan_mock:
+        res = config_flow.get_serial_by_id(sentinel.path)
+        assert res is sentinel.path
+        assert is_dir_mock.call_count == 1
+        assert scan_mock.call_count == 0
+
+
+def test_get_serial_by_id():
+    """Test serial by id conversion."""
+    p1 = patch("os.path.isdir", MagicMock(return_value=True))
+    p2 = patch("os.scandir")
+
+    def _realpath(path):
+        if path is sentinel.matched_link:
+            return sentinel.path
+        return sentinel.serial_link_path
+
+    p3 = patch("os.path.realpath", side_effect=_realpath)
+    with p1 as is_dir_mock, p2 as scan_mock, p3:
+        res = config_flow.get_serial_by_id(sentinel.path)
+        assert res is sentinel.path
+        assert is_dir_mock.call_count == 1
+        assert scan_mock.call_count == 1
+
+        entry1 = MagicMock(spec_set=os.DirEntry)
+        entry1.is_symlink.return_value = True
+        entry1.path = sentinel.some_path
+
+        entry2 = MagicMock(spec_set=os.DirEntry)
+        entry2.is_symlink.return_value = False
+        entry2.path = sentinel.other_path
+
+        entry3 = MagicMock(spec_set=os.DirEntry)
+        entry3.is_symlink.return_value = True
+        entry3.path = sentinel.matched_link
+
+        scan_mock.return_value = [entry1, entry2, entry3]
+        res = config_flow.get_serial_by_id(sentinel.path)
+        assert res is sentinel.matched_link
+        assert is_dir_mock.call_count == 2
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)